lib/src/data/repository/movie_repository.dart

```diff
@@ -12,19 +16,30 @@ class MovieRepository implements IMovieRepository {
   static const String moviesJsonPath = 'mock_data/movies.json';
 
   @override
-  Future<List<Movie>?> getMovies() async {
+  Future<DataState<List<Movie>>> getMovies([
+    int page = 1,
+    MovieEndpoint _ = MovieEndpoint.popular,
+  ]) async {
+    if (movies == null) {
+      try {
+        await _loadMoviesFromJson();
+      } catch (e) {
+        movies = null;
+      }
+    }
```

Comment:
In case reading the files fails for some reason. It's not really necessary since we know that the files are there, but just in case. Setting movies to null is useless though, rewrote it so it tries to load the json and return a data success, and if that fails it returns a data failed